teps/NNNN-catalog-testing.md

```diff
@@ -0,0 +1,495 @@
+Catalog, they need to know how to provide tests that make sure that they work,
+and they need to be provided with infrastructure that they can test against.
+
+### Non-Goals
+
+* This proposal focuses on testing of `Tasks` in the catalog; the test infrastructure
+  would need to be expanded to support the testing of `Pipelines` as well
+* Catalog folder structure defined in [Tekton Catalog Organization TEP](https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md)
+* Catalog support tiers defined in [Tekton Catalog Tiers](https://docs.google.com/document/d/1BClb6cHQkbSpnHS_OZkmQyDMrB4QX4E5JXxQ_G2er7M/edit#heading=h.mfg0tcb14ixk)
+  * TODO: perhaps another TEP for this?
+* Handling (or not) of images required by tasks discussed in [Tekton Catalog Images](https://docs.google.com/document/d/1tkrDWd4Vud0mk5xOiox9JHfg96tjiawrGi0aonA590w/edit)
+* Tekton component compatibility discussed in https://github.com/tektoncd/pipeline/issues/2588
+  and implemented via the label `tekton.dev/minVersion`
+
+
+## Requirements
+
+### Quality
+
+1. If a resource is in the catalog, a user should feel confident that it will work as advertised
+  a. Linting and configuration tests should be applied to all resources
+  b. Testing should be applied regularly to all resources in the catalog
+  c. Folks developing resources should be able to easily run these tests themselves
+  d. It must be possible to apply this testing to resources that rely on external services (e.g. S3, slack)
+  e. Eventually tests should be run against varied Tekton installations, e.g. Tekton on different clouds
+
+### Tekton org external contributions
+ 1. We want to have high quality resources that users can rely on, however we
+   also want to make it possible to for anyone to submit resources to the catalog
+   with very little barrier in order to encourage contributions
+    1. They may initially be true of the catalog and later be true of the hub and not the catalog
+  2. Users should be able to define their own catalogs as well, and use resources from their own
+    catalogs and the Tekton catalog interchangeably
+
+## Proposal
+
+1. Tasks indicate compatibility via `tekton.dev/minVersion` to indicate the min
+  Tekton Pipelines versions they are compatible with
+2. Catalog infrastructure will test **verified** Tasks against:
+  1. The minimum compatible version
+  2. The most recent release
+  3. The nightly release
+3. Verified Tasks have at least one test that can be executed to verify them
+4. Tests are executed against PRs and periodically every evening
+  1. The nightly release test is probably only periodic
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this TEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
```

Comment:
I think you had two good ones in the intro - users should be able to rely on stuff working, and contributors should be able to add things easily.